tests/fsharp/core/nameof/preview/test.fsx

```diff
@@ -343,6 +373,41 @@ do test "lookup name of a generic class"                ((NameOfOperatorForGener
 
 do test "user defined nameof should shadow the operator"(UserDefinedNameOfTests.``user defined nameof should shadow the operator`` ())
 
+do test "can get name of class type parameter"(GenericClassNameOfTests<int>.``can get name of class type parameter`` ())
+do test "can get name of class type parameter"(GenericClassNameOfTests2<FSharp.Data.UnitSystems.SI.UnitSymbols.kg>.``can get name of class unit of measure type parameter`` ())
+do test "can get name of recursive literal"(RecTest.``can get name of recursive literal`` ())
+do test "can get name of literal in recursive module"(RecTest2.``can get name of literal in recursive module`` ())
+
+module PatternMatchingWithNameof =
+    /// Simplified version of EventStore's API
+    type RecordedEvent = { EventType: string; Data: string }
+
+    /// My concrete type:
+    type MyEvent =
+        | A of string
+        | B of string
+
+    let deserialize (e: RecordedEvent) : MyEvent =
+        match e.EventType with
+        | nameof A -> A e.Data
+        | nameof B -> B e.Data
+        | t -> failwithf "Invalid EventType: %s" t
+
+    let getData event =
+        match event with
+        | A amsg -> amsg
+        | B bmsg -> bmsg
+
+    let re1 = { EventType = nameof A; Data = "hello" }
+    let re2 = { EventType = nameof B; Data = "world" }
+
+    let a = deserialize re1
+    let b = deserialize re2
+
+    check "fklwveoihwq1" (getData a) re1.Data
+    check "fklwveoihwq2" (getData b) re2.Data
```

Comment:
I'm not familiar with this bespoke(ish) system for testing. Do these get executed?

Reply:
Yup, it's used throughout tests\fsharp.